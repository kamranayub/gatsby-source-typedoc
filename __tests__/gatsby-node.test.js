const helpers = require('./__fixtures__/helpers');
const { sourceNodes } = require("../gatsby-node");

describe("gatsby-node: sourceNodes", () => {

  describe("typedoc: latest", () => {
    it("should generate project", async () => {
      const nodes = [];

      helpers.actions.createNode.mockImplementation((node) => nodes.push(node));

      await sourceNodes(helpers, {
        src: ["./__fixtures__/simple/index.ts"],
      });

      expect(nodes).toMatchSnapshot();
    });
  });
});
