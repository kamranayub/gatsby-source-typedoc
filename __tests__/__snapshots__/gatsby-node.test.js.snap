// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-node: sourceNodes typedoc: latest should generate project 1`] = `
Array [
  Object {
    "id": "typedoc-default",
    "internal": Object {
      "content": "{\\"id\\":0,\\"name\\":\\"gatsby-source-typedoc\\",\\"kind\\":0,\\"flags\\":{},\\"originalName\\":\\"\\"}",
      "contentDigest": "testdigest",
      "type": "Typedoc",
    },
    "source": Object {
      "categories": undefined,
      "children": undefined,
      "comment": undefined,
      "decorates": undefined,
      "decorators": undefined,
      "flags": Object {},
      "groups": undefined,
      "id": 0,
      "kind": 0,
      "kindString": undefined,
      "name": "gatsby-source-typedoc",
      "originalName": "",
      "sources": undefined,
    },
    "typedocId": "default",
  },
]
`;
